CONTRIBUTING.md

```diff
@@ -0,0 +1,38 @@
+# Contributing to `ACCESS-OM2`s `spack` Environment
+
+## Via Pull Request
+
+### PRs for New Features
+
+For new features, the workflow is fairly straightforward.
+
+Simply open a pull request from your own branch to `main`, in which you make your modifications to the `spack.yaml` (and optionally the `config/versions.json`). The CI/CD will deploy these changes as a `Prerelease` version of `ACCESS-OM2` on `vk83/prerelease`, so you can verify that the changes will work before being merged formally into `main`.
+
+Some things to be mindful of:
+
+* Remember to update the version of the `access-om2` spec in the `spack.yaml` - this is the bit that looks like `- access-om2@git.VERSION`. It is this `VERSION` that will be eventually tagged to the commit.
+* The `config/versions.json` file is used to bundle the deployment with a particular version of `spack-packages` and `spack-config`. Don't forget to change this if there are cool new features in either of the repositories that you want to incorporate into your deployment.
+
+### PRs for Backported Bugfixes
+
+Since we use the `main` branch as a place for the bleeding-edge `ACCESS-OM2` changes, how do we go about backporting bugfixes?
+
+We use dedicated `backport/*.*` branches for bugfixes and additions to past versions of `ACCESS-OM2`. For example, say we have `2024.01.1` version of `ACCESS-OM2` on `main`, which we need to backport a bugfix.
+
+We should:
+
+* Branch off that commit with a `backport/2024.01` branch (if it doesn't already exist)
+* Open a PR off the `backport/2024.01` branch with the fixes, and when it is merged, will be tagged with `2024.01.2` on the `backport/2024.01` branch.
+
+We need to be mindful of the same things as merges of features into `main`:
+
+* Remember to update the version of the `access-om2` spec in the `spack.yaml` - this is the bit that looks like `- access-om2@git.VERSION`. It is this `VERSION` that will be eventually tagged to the commit.
```

Comment:
I kinda wonder if the modifications to the `spack.yaml` (including the versioning) should be left to the developer. I don't know if I like the idea of the environment file being modified on the fly by the CI infrastructure, even if it is for something as small as the version of the `access-om2` package. So we have some options:
* CI infrastructure is allowed to modify the `spack.yaml` as it sees fit. This means that the version stuff needs to be left alone in the `spack.yaml`, because the CI will handle it. This means that we need to have conditions on how the developer edits the file.  
* CI infrastructure is not allowed to modify the `spack.yaml`, but it is allowed to give hints. This would be something like a comments saying "I see you are on a `backport/*.*` branch/I see you added a `type:bugfix` label, please update the version to `X.Y.Z`." We can enforce that this is done in the CI. There could possibly be weirdness with the CI figuring out what major version is next. 
* CI infrastructure is not allowed to modify the `spack.yaml`, and it won't give hints. We leave it all up to the developer to version effectively. Simple, but could lead to errors.

Reply:
The reason I like the idea of the CI taking care of it is that otherwise it places a burden on the developer to understand what is required, and also run commands to determine the current version they're back-porting to. 

So if we leave it to users we'll need to provide a lot of checks to ensure correctness effectively recreating the machinery we would need to do it automatically *and* stuff to report it, when we could just do it automatically straight away.

Reply:
Alright, I get ya. I'll do some more thonking

Reply:
Thanks.

Reply:
After an offline conversation, we noted:
* We will go with a variation of Option 2: CI is not able to modify the `spack.yaml` directly, _without the express permission of the PR creator, through PR commands_. This is a similar thing to https://github.com/ACCESS-NRI/access-om2-configs/blob/main/.github/workflows/pr-2-confirm.yml
* We will go full Clippy and nudge the PR creator when they haven't modified the version of `access-om2` appropriately, where appropriately means 'It isn't the same version as the target branch, and it isn't an existing tag'.